File: data/code_pairs/fortran/930.f90
MODULE types
  TYPE :: tp4
    REAL, PRIVATE :: x = 0.0
    INTEGER, PRIVATE :: i = 0
  END TYPE tp4

  TYPE :: tp3
    REAL :: x = 0.0
    INTEGER, PRIVATE :: i = 0
  END TYPE tp3

  TYPE :: tp2
    TYPE(tp3) :: t
  END TYPE tp2

  TYPE :: tp1
    INTEGER :: i = 0
    TYPE(tp2) :: t
  END TYPE tp1
END MODULE types

MODULE nml
  USE types
  IMPLICIT NONE
  TYPE(tp1) :: t1
  TYPE(tp4) :: t4
  PRIVATE :: i = 0

CONTAINS

  SUBROUTINE y()
    TYPE(tp2) :: y2
    TYPE(tp3) :: y3
    PRINT *, "Subroutine y executed."
  END SUBROUTINE y
END MODULE nml

PROGRAM main
  USE nml
  IMPLICIT NONE

  TYPE :: tp5
    TYPE(tp4) :: t
  END TYPE tp5

  TYPE(tp5) :: t5

  PRINT *, "Main program executed."

CONTAINS

  SUBROUTINE z()
    PRINT *, "Subroutine z executed."
  END SUBROUTINE z

END PROGRAM main

File: data/code_pairs/cpp/930.cpp
#include <iostream>

namespace types {

class tp4 {
private:
    float x = 0.0f;
    int i = 0;

public:
    tp4() = default;
};

class tp3 {
private:
    int i = 0;

public:
    float x = 0.0f;
    tp3() = default;
};

class tp2 {
public:
    tp3 t;
};

class tp1 {
public:
    int i = 0;
    tp2 t;
};

} // namespace types

namespace nml {
using namespace types;

class tp5 {
public:
    tp4 t;
};

tp1 t1;
tp4 t4;

void y() {
    tp2 y2;
    tp3 y3;
    std::cout << "Subroutine y executed." << std::endl;
}

} // namespace nml

int main() {
    using namespace nml;

    tp5 t5;

    std::cout << "Main program executed." << std::endl;

    // Assuming z is somehow called or used
    // z();

    return 0;
}